### DISCUSSION TITLE
lc-power-match-baluns ready to publish

### DISCUSSION THREAD

**Discussion by thePlasm** [dtudk/Admin#5] [Ask for a sub-repo]
I have committed to lc-power-match-baluns, which is now ready for a release. I have merged everything from the draft branch to the main branch (and squashed all of the commits). The draft branch still seems to be the default branch. I am requesting for the main branch to be made the default branch, the draft branch to be deleted, and for the repository to be made public.

Regards,
Rasmus

**Comment by zerothi** [comment]
I have changed default branch and changed to public, I believe you should be able to delete the `draft` branch, it would be instructive for me if you could check it works. :)

**Reply by thePlasm** [reply]
Dear Nick,

Thanks for that. I have checked and the repo seems to work. I have also deleted the draft branch.

I would also like to create a Zenodo record for the repo, though I can't seem to find this repo in the list of repos in Zenodo. I have requested access for Zenodo to access dtudk to create this record.

I also wanted to create a Python package for the repo on PyPI, though I don't think I'll be able to do this in the near future because PyPI has disabled new user registrations.

Regards,
Rasmus

**Reply by thePlasm** [reply]
Could you also please make the description of the repository "Python package and notebooks for power matching LC-baluns"?

**Reply by zerothi** [reply]
You can change the description yourself. I have added zenodo possibility.

**Reply by thePlasm** [reply]
Thanks for your response. When you say that you have added zenodo possibility, does that mean that it will be published when I make a release? Should I add metadata to the zenodo record by means of a CITATION.cff file?

I will perform some final fixes to this repository by next week and make a release when I am done. I will advise once this release has been made.

**Reply by zerothi** [reply]
The zenodo API has access to the `dtudk` namespace. Hence you can use all zenodo functionalities. I would suggest you create a test repo is your own user-space, play with how the citation.cff file should be formatted, and then do the same here.
Zenodo doesn't do anything unless you explicitly tell it too. This is all up to you :)

**Reply by thePlasm** [reply]
Dear Nick,

Thanks for your response.

I have added metadata by means of a `.zenodo.json` file (which I tested in a personal repo) and I have created a release. Do you know how I would be able to find the Zenodo record corresponding to the repo?

Regards,
Rasmus

**Reply by zerothi** [reply]
It is now done, when you make a new release with your repository, you will get an automatic DOI, do expect some delay (could be some days...).

If you want to create a new release of the already released version, you'll have to do it yourself (AFAIK).

**Reply by thePlasm** [reply]
Thanks for letting me know.

Do Zenodo records normally take a few days to publish from GitHub? I looked at a few of the most recent Zenodo records from GitHub and their records seem to have been created a few minutes after they were published. Is there any way for me to check if the Zenodo record has failed to publish? Should I try to create another release in case the webhook did not work?

I have also tried to publish my test repo again on the Zenodo sandbox, which uses the exact same `.zenodo.json` file, which worked.

**Reply by zerothi** [reply]
I have had all experiences, sometimes fast, sometimes slow. :)
It is your decision what to do.

**Reply by thePlasm** [reply]
When I published the test repo to the sandbox, it showed me the record status (i.e. if it was processing, failed or published). Would it be possible for you to check if the record has failed in the dtudk Zenodo?

**Reply by thePlasm** [reply]
Dear Nick, I have made a new release, which has worked. It seems that zenodo did not pick up v1.0.0. I have also published the repo to PyPI.

I have noticed that while the .zenodo.json reflects the collaborators and thesis supervisors accurately, the way that this is presented on the Zenodo record is misleading (incorrect order and duplicate authors). I will confirm with my supervisors regarding how this should appear and adjust the `.zenodo.json` accordingly. Is it possible to amend the metadata of an existing version, or would I need to create a new release to rectify this? Sorry for this mistake and thanks for your help.

**Reply by thePlasm** [reply]
Regarding my previous comment, I have confirmed that the supervisors in the Zenodo record should reflect the updated `.zenodo.json` file (i.e. with the `thesis_supervisors` field removed and with only Jan and Vitaliy shown (in that order)). Would it be possible for the initial release on Zenodo to be rectified to reflect this or would I need to make a new release to amend the record's metadata?

**Reply by zerothi** [reply]
Done, take a look.
One should *always* be able to change metadata, but new/changed files, requires a new release.

**Reply by thePlasm** [reply]
Thanks for your help. I think that is all for publishing this repo. I will close this issue.
